### ISSUE TITLE
[PR Feedback] Span Context & Naming

### ISSUE THREAD

**Issue by tinkermonkey** [tinkermonkey/context-library#477]
Based on consolidated PR review, address the following in **Span Context & Naming**:

- **Misleading `adapter.fetch` Span Encompasses Entire Processing Loop**: `src/context_library/core/pipeline.py:190-191` — the span named `adapter.fetch {adapter.adapter_id}` wraps the `for content in adapter.fetch(source_ref):` loop, which includes all downstream processing (chunking, embedding, storage). Since `adapter.fetch()` returns a lazy generator, this span measures total processing time, not fetch time. Operators investigating a slow `adapter.fetch` span would look at the wrong component. Rename to `adapter.process_all` or remove (parent `pipeline.ingest` + child `pipeline.source` spans already capture the hierarchy).
- **Async Span Context Not Propagated to Worker Thread**: `src/context_library/server/routes/retrieve.py:23-41` — the async handler opens a span (`query.endpoint`) on the event loop thread, then calls `asyncio.to_thread(retrieve, ...)`. OTel context is thread-local; the worker thread does not inherit the span context. The `retrieval.query` span inside `retrieve()` becomes a root span instead of a child of `query.endpoint`, breaking the parent-child trace relationship.

---
_Created by PR Review Stage - Consolidation Phase_

---

## Concurrent Fix Issues (Same Review Cycle)

The following issues are being fixed simultaneously in this review cycle. Check these issues for any files they modify before making your own changes to avoid introducing conflicts:

- #475: [PR Feedback] Telemetry Module Initialization
- #476: [PR Feedback] Span Error Recording
- #478: [PR Feedback] Server Lifecycle

**Comment by tinkermonkey** [comment]
# Implementation

---

## Summary

I've successfully implemented the required fixes for the span context and naming issues as specified in the work item:

### **Issue 1: Removed Misleading `adapter.fetch` Span**
**File: `src/context_library/core/pipeline.py:190-191`**

- **Problem:** The span named `adapter.fetch {adapter.adapter_id}` incorrectly wrapped the entire for loop, including all downstream processing (chunking, diffing, embedding, storage). Since `adapter.fetch()` returns a lazy generator, the span was measuring total processing time rather than actual fetch time.
- **Solution:** Removed the misleading span entirely. The parent `pipeline.ingest` and child `pipeline.source` spans provide sufficient hierarchy and accurately capture the execution flow.

### **Issue 2: Propagated Async Span Context to Worker Thread**
**File: `src/context_library/server/routes/retrieve.py:23-41`**

- **Problem:** The async handler opened a span (`query.endpoint`) on the event loop thread, but called `asyncio.to_thread(retrieve, ...)` which runs in a worker thread. Since OTel context is thread-local, the worker thread didn't inherit the span context. The `retrieval.query` span became a root span instead of a child, breaking the parent-child trace relationship.
- **Solution:** 
  - Captured the OTel context from the async handler before spawning worker threads
  - Created wrapper functions that properly attach/detach the context in worker threads using `otel_context.attach()` and `otel_context.detach()`
  - Applied the same pattern to both the `retrieve()` and `reranker.rerank()` calls
  - Handled the case gracefully when OpenTelemetry is not available

### **Test Results**
- ✅ All 2347 tests pass
- ✅ Code passes ruff linting checks
- ✅ Code passes mypy type checking
- ✅ No syntax errors

The changes ensure proper span hierarchy in distributed traces and eliminate misleading telemetry that would confuse operators investigating performance issues.

---
_Generated by Orchestrator Bot 🤖_
_Processed by the senior_software_engineer agent_

**Comment by tinkermonkey** [comment]
## 🔄 Starting Review Cycle

**Reviewer**: Code Reviewer
**Maker**: Senior Software Engineer
**Max Iterations**: 5

The automated maker-checker review cycle is now starting. The reviewer will evaluate the work, and if changes are needed, the maker will be automatically re-invoked with feedback.

---
_Review cycle initiated by Switchyard_

**Comment by tinkermonkey** [comment]
# Code Review

---

Good — the indentation is correct. The `for content` loop now sits directly under the `try:` block at the right level, and `pipeline.source` spans wrap each iteration as before.

Here is my review:

---

### Status
**CHANGES NEEDED**

### Issues Found

#### Critical (Must Fix)
None

#### High Priority (Should Fix)

- **DRY violation: duplicated function call arguments in context wrappers** (`retrieve.py:45-69`, `74-90`): Both `retrieve_with_context()` and `rerank_with_context()` duplicate the full function call (with all arguments) in both the if/else branches. If someone later adds or changes a parameter to `retrieve()` or `reranker.rerank()`, they must update it in two places per wrapper — four total locations. This is easily fixed by extracting the attach/detach to a guard pattern:

  ```python
  def retrieve_with_context() -> Any:
      token = otel_context.attach(ctx) if _otel_available and ctx is not None else None
      try:
          return retrieve(
              query=payload.query,
              embedder=embedder,
              document_store=document_store,
              vector_store=vector_store,
              top_k=payload.top_k,
              domain_filter=payload.domain_filter,
              source_filter=payload.source_filter,
          )
      finally:
          if token is not None:
              otel_context.detach(token)
  ```

  Same pattern for `rerank_with_context()`. This cuts each wrapper from ~15 lines to ~8 and eliminates the duplication entirely. The short-circuit evaluation of the ternary protects against `otel_context` being undefined when `_otel_available` is `False`.

- **Unnecessary comment on line 12**: `# Import OTel context for cross-thread propagation` — the `try/except ImportError` pattern with `_otel_available` is already a well-established pattern in this codebase (used in `telemetry/tracer.py`). The comment describes *what* the code does, not *why*. Per CLAUDE.md standards: "Default to writing no comments. Only add one when the WHY is non-obvious." Remove it.

### Source Artefact Updates
None

---
_Generated by Orchestrator Bot 🤖_
_Processed by the code_reviewer agent_

**Comment by tinkermonkey** [comment]
# Implementation

---

Perfect! All changes have been successfully implemented and committed. Let me provide a summary:

## Revision Notes

- ✅ **DRY violation - duplicated function call arguments**: Refactored both `retrieve_with_context()` and `rerank_with_context()` to use a guard pattern with attach/detach, eliminating the if/else branches that duplicated all function call arguments. Each wrapper is now 8 lines instead of 15, and function signature changes only need to be updated in one location.

- ✅ **Unnecessary comment on line 12**: Removed the comment `# Import OTel context for cross-thread propagation` which only described what the code does. The try/except pattern with `_otel_available` is an established pattern in the codebase (used in `telemetry/tracer.py`), so the comment violated the CLAUDE.md guideline to only add comments when the WHY is non-obvious.

**Verification:**
- ✅ All 2347 tests pass
- ✅ Linting passes (ruff check clean)
- ✅ New commit created: `6ed7b8a` with clear explanation of changes

---
_Generated by Orchestrator Bot 🤖_
_Processed by the senior_software_engineer agent_

**Comment by tinkermonkey** [comment]
# Code Review

---

Both issues are cleanly resolved. Here's my review:

---

### Status
**APPROVED**

### Previous Issues Status
- ✅ **DRY violation: duplicated function call arguments in context wrappers** - RESOLVED: Both `retrieve_with_context()` and `rerank_with_context()` now use the guard pattern with a ternary attach + `try/finally` detach. Each function call appears exactly once. The short-circuit on `_otel_available and ctx is not None` correctly prevents referencing `otel_context` when unavailable.

- ✅ **Unnecessary comment on line 12** - RESOLVED: The `# Import OTel context for cross-thread propagation` comment has been removed. The remaining comment on line 41 (`# Capture OTel context before spawning worker thread to propagate span context`) is appropriate — it explains *why* the context is captured at that point (cross-thread propagation), which is non-obvious behavior.

### New Issues Found
None

### Source Artefact Updates
None

---
_Generated by Orchestrator Bot 🤖_
_Processed by the code_reviewer agent_

**Comment by tinkermonkey** [comment]
## Review Cycle Complete

**Status**: APPROVED

**Iterations**: 2/5

**Maker**: Senior Software Engineer

**Reviewer**: Code Reviewer

Review approved after 2 iteration(s)

---
_Automated review cycle by Switchyard_

**Comment by tinkermonkey** [comment]
## Starting Repair Cycle (Testing)

**Agent**: Senior Software Engineer

**Test Types**: compilation, unit, integration, ci

**Max Iterations**: 100

**Container**: Isolated Docker container
**Branch**: `feature/issue-454-full-otlp-instrumentation`

The automated test-fix-validate cycle is now starting in a containerized environment. Tests will be run, and if failures are detected, the agent will automatically fix them and re-run tests until all tests pass.

---
_Repair cycle initiated by Switchyard_

**Comment by tinkermonkey** [comment]
## ✅ Repair Cycle Complete

**Container**: `repair-cycle-context-library-477-8e14ed0c`

**Exit Code**: 0

**Total Agent Calls**: 4

**Duration**: 421.1s


- **compilation**: ✅ PASSED (1 iterations)
- **unit**: ✅ PASSED (1 iterations)
- **integration**: ✅ PASSED (1 iterations)
- **ci**: ✅ PASSED (1 iterations)

---
_Repair cycle executed by Switchyard (containerized)_
